app/services/reaction_handler.rb

```diff
@@ -89,6 +89,7 @@ def create_new_reaction
       rate_limit_reaction_creation
       sink_articles(reaction)
       send_notifications(reaction)
+      update_last_reacted_at(reaction)
```

Comment:
`last_reacted_at` was only updated when a user went to `http://localhost:3000/readinglist` and toggled the archive state of an item on the reading list. 

https://github.com/forem/forem/blob/c24e227f559dc42aa0c259db62404950aca0063c/app/javascript/readingList/readingList.jsx#L112-L115

https://github.com/forem/forem/blob/c24e227f559dc42aa0c259db62404950aca0063c/app/controllers/reading_list_items_controller.rb#L13

Hence, I'd update the behavior of last_reacted_at to better reflect what the name implies it does as per https://github.com/forem/forem/issues/18796#issuecomment-1397195747.


I could use a second pair of eyes on the following: 
- We were using `last_reacted_at` in some places as cache keys and now that last_reacted_at will be updated when a user creates a reaction, I want to make sure that it won't break or make the caching less accurate.
- I only update `last_reaction_at` when a user "activates" a reaction and not when they "deactivate" it, so for example, when a user 'likes' a post but not when they 'unlike' it. I was wondering what folks thoughts are on that?

Reply:
I think with the cache stuff, as you pointed out, there's some redundancy here, where this probably doesn't have a big meaningful impact. Definitely the right thing to think about, but I think we're probably fine with existing cache keys not getting thrown too much.

I think the logic of only updating when a reaction is activated is appropriate. The logic we're storing at is the last time they created a reaction, and this is only used for cache keys and query rankings, etc. so I don't think there is a perfect answer for how it _should_ behave given this kind of question. IMO the logic you've chosen is intuitive enough and reasonable.

I think you made the right choices here overall. 👍